iarrc/src/color_detector/color_detector.cpp

```diff
@@ -40,12 +38,11 @@ namespace iarrc {
             return;
         }
 
-        const Mat &frame = cv_ptr->image;
-
-        Mat output = Mat::zeros(frame.rows, frame.cols, CV_8UC3);
-
-        const Mat &frame_masked = frame(mask);
-
+        const Mat &frameBGR = cv_ptr->image;
+        const Mat frameHSV = Mat::zeros(frameBGR.rows, frameBGR.cols, CV_8UC3);
+        cvtColor(frameBGR, frameHSV, CV_BGR2HSV);
```

Comment:
Minor comment for possible future work. Depending on the size of the mask, reversing the order of these operations may give better performance. Calling cvtColor on the masked frame will have to convert fewer pixels between color spaces.